sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilderUnitTest.java

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.textanalytics;
+
+import org.junit.jupiter.api.Test;
+
+import static com.azure.ai.textanalytics.TestUtils.VALID_HTTPS_LOCALHOST;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/**
+ * Unit tests for Text Analytics client builder
+ */
+public class TextAnalyticsClientBuilderUnitTest {
+
+    /**
+     * Test for missing endpoint
+     */
+    @Test
+    public void missingEndpointAsyncClient() {
+        assertThrows(NullPointerException.class, () -> {
+            final TextAnalyticsClientBuilder builder = new TextAnalyticsClientBuilder();
+            builder.buildAsyncClient();
+        });
+    }
+
+    /**
+     * Test for missing endpoint
+     */
+    @Test
+    public void missingEndpoint() {
+        assertThrows(NullPointerException.class, () -> {
+            final TextAnalyticsClientBuilder builder = new TextAnalyticsClientBuilder();
+            builder.buildClient();
+        });
+    }
+
+    /**
+     * Test for invalid endpoint
+     */
+    @Test
+    public void invalidProtocol() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            final TextAnalyticsClientBuilder builder = new TextAnalyticsClientBuilder();
+            builder.endpoint(TestUtils.INVALID_URL);
+        });
+    }
+
+    /**
+     * Test for null API key
+     */
+    @Test
+    public void nullApiKey() {
```

Comment:
Should there be a test for empty API Key ?  what is some one provide .apiKey("")  ,  should that  work ?

Reply:
an empty api key test added